test/test_linalg.py

```diff
@@ -3667,22 +3679,26 @@ def test_matrix_rank_atol_rtol(self, device, dtype):
         # test float and tensor variants
         for tol_value in [0.81, torch.tensor(0.81, device=device)]:
             # using rtol (relative tolerance) takes into account the largest singular value (1.5 in this case)
-            result = torch.linalg.matrix_rank(a, rtol=tol_value)
+            result = matrix_rank(a, rtol=tol_value)
             self.assertEqual(result, 2)  # there are 2 singular values above 1.5*0.81 = 1.215
 
             # atol is used directly to compare with singular values
-            result = torch.linalg.matrix_rank(a, atol=tol_value)
+            result = matrix_rank(a, atol=tol_value)
             self.assertEqual(result, 7)  # there are 7 singular values above 0.81
 
             # when both are specified the maximum tolerance is used
-            result = torch.linalg.matrix_rank(a, atol=tol_value, rtol=tol_value)
+            result = matrix_rank(a, atol=tol_value, rtol=tol_value)
             self.assertEqual(result, 2)  # there are 2 singular values above max(0.81, 1.5*0.81)
 
     @skipCUDAIfNoMagma
     @skipCPUIfNoLapack
-    @dtypes(*floating_and_complex_types())
-    def test_matrix_rank_empty(self, device, dtype):
-        matrix_rank = torch.linalg.matrix_rank
+    @dtypes(torch.float64)
+    def test_matrix_rank_atol_rtol(self, device, dtype):
+        self._test_matrix_rank_atol_rtol(device, dtype)
```

Comment:
This one just forward as well, delete.

Reply:
You need this forward. You cannot simply introduce a test function with custom parameters. All `test_*` functions are processed with Python magic.

Reply:
But simply rename the underscore function to `def test_matrix_rank_atol_rtol(self, device, dtype):` and remove this forward. Same in all the other occurrences

Reply:
With that I would need to copy the test suite written for `matrix_rank`. The goal is to reuse the code written for `matrix_rank`.

Reply:
Why?

Reply:
No, you cannot call `test_*` functions because these are dynamically modified. I would need to at least wrap them in all the decorators.

Reply:
Offline discussion: These things are necessary due to how `pytest` works. @nikitaved will leave comments saying that we're doing this because they are used in `linalg.matrix_rank` and this function.